src/compiler/checker.ts

```diff
@@ -16009,13 +16009,14 @@ namespace ts {
                     unmatchedProperty.valueDeclaration
                     && isNamedDeclaration(unmatchedProperty.valueDeclaration)
                     && isPrivateIdentifier(unmatchedProperty.valueDeclaration.name)
+                    && source.symbol.valueDeclaration
                     && isClassDeclaration(source.symbol.valueDeclaration)
                 ) {
                     const privateIdentifierDescription = unmatchedProperty.valueDeclaration.name.escapedText;
                     const symbolTableKey = getSymbolNameForPrivateIdentifier(source.symbol, privateIdentifierDescription);
-                    if (symbolTableKey && !!getPropertyOfType(source, symbolTableKey)) {
+                    if (symbolTableKey && getPropertyOfType(source, symbolTableKey)) {
                         const sourceName = source.symbol.valueDeclaration.name;
-                        const targetName = isClassDeclaration(target.symbol.valueDeclaration) ? target.symbol.valueDeclaration.name : undefined;
+                        const targetName = target.symbol.valueDeclaration && isClassDeclaration(target.symbol.valueDeclaration) ? target.symbol.valueDeclaration.name : undefined;

```

Comment:
```suggestion
                        const targetName = getDeclarationName(target.symbol.valueDeclaration);
```

(and for sourceName too)

Reply:
you'll have to change the type of `diagnosticName` to `(nameArg: __String | DeclarationName) => string`, since it currently has a too-narrow type.